src/changelog.ts

```diff
@@ -383,11 +383,13 @@ export default class Changelog {
 
     const unreleasedChanges = this.#changes[unreleased];
 
-    for (const category of Object.keys(unreleasedChanges) as ChangeCategory[]) {
+    for (const category of Object.keys(
+      unreleasedChanges,
+    ) as (keyof typeof unreleasedChanges)[]) {
       if (releaseChanges[category]) {
         releaseChanges[category] = [
-          ...(unreleasedChanges[category] as string[]),
-          ...(releaseChanges[category] as string[]),
+          ...(unreleasedChanges[category] ?? []),
+          ...(releaseChanges[category] ?? []),
```

Comment:
`@metamask/utils` now has a `getKnownPropertyNames` function which is specifically designed to iterate over the properties of an enum, while retaining types. This should obviate the need for the type assertion here at least. Note that this requires adding `@metamask/utils` as a dependency, but I think that's okay (we use it everywhere):

``` suggestion
    for (const category of getKnownPropertyNames(unreleasedChanges)) {
      if (releaseChanges[category]) {
        releaseChanges[category] = [
          ...(unreleasedChanges[category] ?? []),
          ...(releaseChanges[category] ?? []),

Reply:
This is really neat! Thanks for the pointer.